### PR TITLE
Feature: Add recommender tests

### PR DESCRIPTION
**Pull request by Bilelkihal**
### Done in this PR:
In this pull request, I've incorporated end-to-end (E2E) system tests for the recommender page across various use cases and scenarios.

Technically, I've opted not to rely on the current production API for fetching data. Instead, I've implemented mock responses for all API calls specific to this page. This ensures that our tests solely focus on the UI, and they remain effective even if we modify elements in the database or make other changes.

### Demo
[Capture vidéo du 26-04-2024 11:47:46.webm](https://github.com/ontoportal-lirmm/bioportal_web_ui/assets/61744974/2292cea2-4665-4fa6-94bb-90fb977dd520)

### CODE REVIEW

File: test/fixtures/recommender.yml [hunk 1]
```diff
@@ -0,0 +1,973 @@
...
+        notes: https://data.agroportal.lirmm.fr/ontologies/EFO/notes
+        groups: https://data.agroportal.lirmm.fr/ontologies/EFO/groups
+        categories: https://data.agroportal.lirmm.fr/ontologies/EFO/categories
+        latest_submission: https://data.agroportal.lirmm.fr/ontologies/EFO/latest_submission
+        projects: https://data.agroportal.lirmm.fr/ontologies/EFO/projects
+        download: https://data.agroportal.lirmm.fr/ontologies/EFO/download
+        views: https://data.agroportal.lirmm.fr/ontologies/EFO/views
+        analytics: https://data.agroportal.lirmm.fr/ontologies/EFO/analytics
+        ui: http://agroportal.lirmm.fr/ontologies/EFO
+        "@context":
```
Comment (syphax-bouazzouni) [inline test/fixtures/recommender.yml:29]: no need to have the `@context` and `@links`,  remove it, will make this file smaller

File: test/test_helper.rb [hunk 2]
```diff
@@ -37,7 +37,11 @@ def load_all_fixtures
   fixture_files.each do |fixture_file|
     fixture_name = File.basename(fixture_file, '.yml')
     data = YAML.load_file(fixture_file)
-    fixtures_data[fixture_name] = OpenStruct.new(Array(data).map{|key, hash| [key , OpenStruct.new(hash)]}.to_h)
+    if data["yaml_structure"].eql?("regular")
```
Comment (syphax-bouazzouni) [inline test/test_helper.rb:40]: why this ?

File: test/system/recommender_page_test.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                        'Host'=>'localhost:9393',
+                        'User-Agent'=>'NCBO API Ruby Client v0.1.0'
+                })
+            .to_return(status: 200, body: @sample_response.to_json, headers: {})
+            
+        WebMock.stub_request(:get, @ontologies_url)
+            .with(
+              headers: {
+                    'Accept'=>'application/json',
+                    'Authorization'=>'apikey token=1de0a270-29c5-4dda-b043-7c3580628cd5',
```
Comment (syphax-bouazzouni) [inline test/system/recommender_page_test.rb:59]: make the `apikey` and `host` be the configuration variables

File: test/system/recommender_page_test.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+        find("div.insert-sample-text-button").find("div.button").click
+
+        WebMock.enable!
+        WebMock.stub_request(:post, @recommender_api)
+            .with(
+                body: "{\"input\":\"Melanoma is a malignant tumor of melanocytes found mainly in the skin but also in the intestine and the eye.\",\"input_type\":\"1\",\"output_type\":\"1\",\"wc\":\"0.55\",\"wa\":\"0.15\",\"wd\":\"0.15\",\"ws\":\"0.15\",\"max_elements_set\":\"3\",\"controller\":\"recommender\",\"action\":\"index\"}",
+                headers: {
+                        'Accept'=>'application/json',
+                        'Authorization'=>'apikey token=1de0a270-29c5-4dda-b043-7c3580628cd5',
```
Comment (syphax-bouazzouni) [inline test/system/recommender_page_test.rb:48]: make the `apikey` and `host` be the configuration variables

File: test/system/recommender_page_test.rb [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                    'Accept'=>'application/json',
+                    'Authorization'=>'apikey token=1de0a270-29c5-4dda-b043-7c3580628cd5',
+                    'Host'=>'localhost:9393',
+                    'User-Agent'=>'NCBO API Ruby Client v0.1.0'
+              })
+            .to_return(status: 200, body: ([]).to_json, headers: {})
+
+        
+        find(@recommender_submit_button).click
+
```
Comment (syphax-bouazzouni) [inline test/system/recommender_page_test.rb:67]: you don't check the results? for example check the number of rows of the table is correct.

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]

File: test/test_helper.rb (hunk 2 above)
Comment (Bilelkihal) [reply test/test_helper.rb:40]: This OpenStruct.new doesn't work with the structure of the result of the recommender, so I need to return it as it is

**Review by syphax-bouazzouni — APPROVED** [review]

**Pull request merged**